Sources/NIO/EventLoopFuture.swift

```diff
@@ -217,6 +217,32 @@ public struct EventLoopPromise<Value> {
         self._resolve(value: result)
     }
 
+    /// Execute the given function and complete the promise.
+    ///
+    /// Specifically, the function is executed synchronously on the `EventLoop`
+    /// that was used to create the promise.
+    ///
+    /// - parameters:
+    ///     - body: The function to execute.
+    @inlinable
+    public func completeWith(_ body: @escaping () throws -> Value) {
```

Comment:
A minor note: overloading based on argument type is a bit tricky. It might be nice to give this a different name: `completeWithBlock` or similar might be good.

Reply:
Thank you for the review!! `completeWithBlock` does look better, I just change it accordingly.